### DISCUSSION TITLE
Keyball39の特殊キーコードについて

### DISCUSSION THREAD

**Discussion by UGKEN** [Yowkees/keyball#129] [Q&A] [answered]
https://github.com/Yowkees/keyball/blob/main/keyball39/doc/rev1/buildguide_jp.md
上記ビルドガイドの末尾に記述されておりますKeyball用の特殊キーコードをリマップしてみたのですが、当方が使用しておりますKeyball39では、どうもビルドガイドに記述されたキーコードと若干異なるようにみえます。
ざっと調べてみたところ、下記のキーコードがビルドガイドの記述と異なっているようでした。
（左がビルドガイドの記述：右が実際にその機能を持ったコード）

- CPIを100増加...0x5DA7 : 0x5DA8
- CPIを100減少...0x5DA8 : 0x5DA9
- CPIを1000増加...0x5DA9 : 0x5DAA
- CPIを1000減少...0x5DAA : 0x5DAB
- スクロール除数を１つ上げます...0x5DAD : 0x5DAE
- スクロール除数を１つ下げます...0x5DAE : 0x5DAF

これについて原因が不明なのですが、当方の環境等になにか問題がありますでしょうか？

**Comment by Yowkees** [comment, marked as answer]
返信が遅くなり失礼しました。
公開しているKeyball用のファームウェアのビルドではQMKの0.16.3を使ってますので、異なるバージョンを使ってるとズレる可能性があります。
Remapで公開しているファームや、下記GitHub Releasesに上がってるものを使っていただければと思います。
https://github.com/Yowkees/keyball/releases/tag/keyball61%2Fv1.1.1

**Reply by UGKEN** [reply]
Yowkees様
ご返信ありがとうございます。
下記ページにある「Keyball39_via（July 29th 2022, 14:41:26 | SHA256: 2c330a8e8684cca835f22885eb76756aa71b45c0d5157f4b8ef4687fba5dbc6d）」のファームウェアをL側/R側両方にFLASHしてSUCCESSしたのですが、L側/R側どちらにUSBケーブルを接続した場合でも症状には特に変化ありませんでした。
https://remap-keys.app/catalog/k895xiCpsM5zlYZCPTEN/firmware

ですが、とりあえずキーコードを1ずつずらしてマッピングすることで問題なくCPI等は調整できており、現状で特に困っていないため、このまま運用したいと思います。
